ipn/ipnlocal/local.go

```diff
@@ -2257,13 +2257,13 @@ func (b *LocalBackend) shouldUploadServices() bool {
 // changed.
 //
 // On non-multi-user systems, the uid should be set to empty string.
-func (b *LocalBackend) SetCurrentUserID(uid string) {
+func (b *LocalBackend) SetCurrentUserID(uid ipnauth.WindowsUserID) {
 	b.mu.Lock()
-	if b.pm.CurrentUserID() == uid {
+	if b.pm.CurrentUserID() == string(uid) {
 		b.mu.Unlock()
 		return
 	}
-	if err := b.pm.SetCurrentUserID(uid); err != nil {
+	if err := b.pm.SetCurrentUserID(string(uid)); err != nil {
```

Comment:
Make this also take WindowsUserID?

Reply:
Done. It ended up having to move the type to ipn instead of ipnauth but that seems fine. Otherwise `ipn` starts growing weird deps.